ert_shared/client/session.py

```diff
@@ -0,0 +1,58 @@
+import os
+import requests
+import json
+
+from urllib.parse import urljoin
+from typing import Any, Tuple, Optional
+
+
+class Session(requests.Session):
+    def __init__(self):
+        super().__init__()
+        self._base_url: Optional[str] = None
+        self._headers: Optional[Tuple] = ()
+
+        # Should the storage_server.json also be used?
+        # If so is there order of prefence
+        if "ERT_STORAGE_CONNECTION_STRING" in os.environ:
+            try:
+                self._connection_info = json.loads(
+                    os.environ["ERT_STORAGE_CONNECTION_STRING"]
+                )
+            except json.JSONDecodeError:
+                raise RuntimeError("Invalid Storage Connection configuration")
+
+        if {"urls", "authtoken"} <= self._connection_info.keys():
+            self._base_url = self.__resolve_url()
+            self._headers = {"Token": self._connection_info["authtoken"]}
+        else:
+            raise RuntimeError("Invalid Storage Connection configuration")
+
+    def __enter__(self):
+        # Do health check?
+        # Seems redudant
+        return self
+
+    def __exit__(self, *args):
+        # No real cleanup todo?
+        pass
+
+    def __resolve_url(self) -> str:
```

Comment:
We tend to use single underscores for "private" methods
```suggestion
    def _resolve_url(self) -> str:
```

Reply:
Noted for future